### DISCUSSION TITLE
방장과 부방장을 알려면 이렇게 하는게 아닐까요..?

### DISCUSSION THREAD

**Discussion by je0404** [storycraft/node-kakao#483] [Q&A]
```
const Kakao = require('node-kakao');
const DEVICE_UUID = '';
const DEVICE_NAME = '';

const EMAIL = '';
const PASSWORD = ''

const CLIENT = new Kakao.TalkClient();

CLIENT.on('chat', (data, channel) => {
const sender = data.getSenderInfo(channel);
if(sender.perm === OpenChannelUserPerm.MANAGER || sender.perm === OpenChannelUserPerm.OWNER){ 
   return;
  }else{
        if(data.text.includes("ㅅㅂ"){
            channel.hideChat(data.chat).then().catch();
          }
}
});
```

방장 또는 부방장이 아닌 사람이 욕설을하면 가리게 하려는데
에러 뜨는 건 없지만 작동을 안해서요 뭔가 잘못된게 있나요?

**Comment by raravel** [comment]
#400

**Reply by raravel** [reply]
디버깅 하시려면 sender 정보는 알맞게 오는지 출력, perm에 데이터가 있는지 출력
data.text에 채팅데이터가 있는지 출력

이런 식으로 하나하나 디버깅하셔야 합니다.

또한 첨부하신 소스가 저게 끝이라면 OpenChannelUserPerm 은 undefined겠네요.

**Comment by dalgkdbs** [comment]
const Kakao = require('node-kakao');
const DEVICE_UUID = '';
const DEVICE_NAME = '';

const EMAIL = '';
const PASSWORD = ''

const CLIENT = new Kakao.TalkClient();

const banWord = ['ㅅㅂ', 'ㅂㅅ', 'ㅄ'];

CLIENT.on('chat', (data, channel) => {
const sender = data.getSenderInfo(channel);
if (sender.perm === Kakao.OpenChannelUserPerm.NONE){ 
    const find = banWord.find(s => data.text.includes(s));
    if (find) {
      channel.hideChat(data.chat).then().catch();
      channel.sendChat(`${sender.nickname}님 욕설 사용 : ${find}`);
    } 
  }
});

**Reply by je0404** [reply]
감사합니다 ! 혹시 data.text.includes(s)); 이부분
data.text가 아니라 다른 타입으로 쓸 수 도있나요??
const find = banWord.find(s => data.chat.type==3.includes(s));

**Reply by dalgkdbs** [reply]
originalType를 비교하시려는거면
따로 type목록을 배열로 선언해두시고
array.includes(data.chat.type)  로 체크하시면 됩니다.

**Reply by dalgkdbs** [reply]
const banTypes = [71];
if (banTypes.includes(data.chat.type)) {
    // todo
}
